p2p/protocols/accounting.go

```diff
@@ -119,43 +123,33 @@ func SetupAccountingMetrics(reportInterval time.Duration, path string) *Accounti
 }
 
 // Send takes a peer, a size and a msg and
-//   - calculates the cost for the local node sending a msg of size to peer querying the message for its price
 //   - credits/debits local node using balance interface
-func (ah *Accounting) Send(peer *Peer, size uint32, msg interface{}) error {
-	// get the price for a message
-	var pricedMessage PricedMessage
-	var ok bool
-	// if the msg implements `Price`, it is an accounted message
-	if pricedMessage, ok = msg.(PricedMessage); !ok {
-		return nil
-	}
-	// evaluate the price for sending messages
-	costToLocalNode := pricedMessage.Price().For(Sender, size)
+func (ah *Accounting) Apply(peer *Peer, costToLocalNode int64, size uint32) error {
 	// do the accounting
 	err := ah.Add(costToLocalNode, peer)
 	// record metrics: just increase counters for user-facing metrics
 	ah.doMetrics(costToLocalNode, size, err)
 	return err
 }
 
-// Receive takes a peer, a size and a msg and
-//   - calculates the cost for the local node receiving a msg of size from peer querying the message for its price
-//   - credits/debits local node using balance interface
-func (ah *Accounting) Receive(peer *Peer, size uint32, msg interface{}) error {
+//   - calculates the cost for the local node sending a msg of size to peer querying the message for its price
+func (ah *Accounting) Validate(peer *Peer, size uint32, msg interface{}, payer Payer) (int64, error) {
```

Comment:
maybe add a comment here about what the `int64` result means/does, since it's not immediately obvious based on the function name